osu.Game.Rulesets.Osu.Tests/TestSceneSliderSnaking.cs

```diff
@@ -0,0 +1,304 @@
+            config.BindWith(OsuRulesetSetting.SnakingInSliders, snakingIn);
+            config.BindWith(OsuRulesetSetting.SnakingOutSliders, snakingOut);
+        }
+
+        private DrawableSlider slider;
+        private DrawableSliderRepeat repeat;
+        private Vector2 vector;
+
+        [SetUpSteps]
+        public override void SetUpSteps() { }
+
+        [Test]
+        public void TestSnaking()
+        {
+            AddStep("have autoplay", () => autoplay = true);
+            base.SetUpSteps();
+            AddUntilStep("wait for track to start running", () => track.IsRunning);
+
+            AddStep("retrieve 1st slider", () => slider = (DrawableSlider)Player.DrawableRuleset.Playfield.AllHitObjects.First());
+            testLinear(true);
+            testLinear(false);
+            AddStep("retrieve 2nd slider", () => slider = (DrawableSlider)Player.DrawableRuleset.Playfield.AllHitObjects.Skip(1).First());
+            testRepeating(true);
+            testRepeating(false);
+            AddStep("retrieve 3rd slider", () => slider = (DrawableSlider)Player.DrawableRuleset.Playfield.AllHitObjects.Skip(2).First());
+            testDoubleRepeating(true);
+            testDoubleRepeating(false);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestArrowStays(bool isHit)
+        {
+            var isSame = isHit ? "is same" : "decreased";
+            var enable = isHit ? "enable" : "disable";
+
+            AddStep($"{enable} autoplay", () => autoplay = isHit);
+            setSnaking(true);
+            base.SetUpSteps();
+
+            addSeekStep(13500);
+            AddStep("retrieve 2nd slider repeat", () =>
+            {
+                var drawable = Player.DrawableRuleset.Playfield.AllHitObjects.Skip(1).First();
+                repeat = drawable.ChildrenOfType<Container<DrawableSliderRepeat>>().First().Children.First();
+            });
+            AddStep("Save repeat vector", () => vector = repeat.Position);
+            addSeekStep(13700);
+            AddAssert($"Repeat vector {isSame}", () => isHit ? Precision.AlmostEquals(vector.X, repeat.X, 1) && Precision.AlmostEquals(vector.Y, repeat.Y, 1) : repeat.X < vector.X && repeat.Y < vector.Y);
+        }
+
+        private void testLinear(bool snaking)
+        {
+            var increased = snaking ? "increased" : "is same";
+
+            setSnaking(snaking);
+            addSeekStep(1800);
+            AddStep("Save end vector", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                vector = body.CurrentCurve.Last();
+            });
+            addSeekStep(1900);
+            AddAssert($"End vector {increased}", () =>
+            {
+                var body = (PlaySliderBody)slider.Body.Drawable;
+                var last = body.CurrentCurve.Last();
+                return snaking ? last.X > vector.X && last.Y > vector.Y : last == vector;
```

Comment:
Why this is using strict equality and not `Precision.AlmostEquals` as above?

Reply:
body path vectors don't update at all if snaking is turned off. repeat arrow position has a chance to update depending on when in frame hit is registered.

Reply:
I'd make that a comment as it's not immediately obvious.